app/src/main/java/nl/tudelft/pixelperfect/client/ClientListener.java

```diff
@@ -51,13 +52,13 @@ public void messageReceived(Client source, Message message) {
         } else if (message instanceof RoleChosenMessage) {
             RoleChosenMessage roleMessage = (RoleChosenMessage) message;
             switch(roleMessage.getRole()) {
-                case 0:
+                case GUNNER :
                     RoleActivity.updateButtons(0);
                     break;
```

Comment:
SUGGESTION: while this is good enough (imo), consider moving the updateButton method to each respective Enum and just call roleMessage.getRole().updateButton() to avoid the switch all together.